File: workflow_shell/git/force-push.py
import click
from ..util import get_result
from ..util import run


command_string = "gpfo"
command_help = "Force pushes the current branch into a remote branch of the same name"
short_help = "Git - force pushes to origin"


def main(*args, **kwargs):
    branch_name = get_result(["git", "rev-parse", "--abbrev-ref", "HEAD"])
    run(["git", "push", "-f", "origin", branch_name])

File: src/workflow_shell/commands/git/force_push.clj
(ns workflow-shell.commands.git.force_push
  (:require [clojure.java.shell :refer [sh]])
  (:gen-class))

(def gpfo (hash-map
           :description "Force pushes the current branch into a remote branch of the same name"
           :execute (fn [& args]
                      (let [branch-name (clojure.string/trim (:out (sh "git" "rev-parse" "--abbrev-ref" "HEAD")))
                            result  (sh "git" "push" "-f" "origin" branch-name)]
                        (println result)
                        (if-not (empty? (:out result))
                          (println (:out result))
                          (println (:err result)))))))